### PR TITLE
use encoding/htmlindex package for finding decoder

### PR DESCRIPTION
**Pull request by urso**
encoding/htmlindex package provides rather exhausitve list of known encodings
in comparison to harvester/encoding.go table.

See list of known encodings:
https://github.com/golang/text/blob/master/encoding/htmlindex/tables.go#L94

### CODE REVIEW

**Comment by ruflin** [comment]
Should we link to http://www.w3.org/TR/encoding/ in the docs for the encoding list so we only list the most important ones?

**Comment by monicasarbu** [comment]
LGTM

File: harvester/encoding.go [hunk 1]
```diff
@@ -22,29 +20,16 @@ var encodings = map[string]Decoder{
 	"nop":   Plain,
 	"plain": Plain,
 
-	// utf8 (validate input)
-	"utf-8": trans(encoding.UTF8Validator),
+	// utf8 (validate input) - shadow htmlindex utf8 codecs not validating input
+	"unicode-1-1-utf-8": trans(encoding.UTF8Validator),
```
Comment (ruflin) [inline harvester/encoding.go]: why do we use the same utf-8 three times?

**Comment by ruflin** [comment]
LGTM, needs rebase. Lets also add tests for some more encodings.

File: harvester/encoding.go (hunk 1 above)
Comment (urso) [inline harvester/encoding.go]: they are the same, but with different name. These utf-8 entries shadow the utf-8 entries in htmlindex, which is equivalent to plain (no error checks)

**Pull request merged**